docs/VMR-Feature-Band-Support-Work-Plan.md

```diff
@@ -0,0 +1,136 @@
+
+❓ Open Question:
+There will need to be some way to detect when we are in one of these 2xx branches. What mechanism should be used for that? Maybe checking `packageVersion` of the sdk in `source-manifest.json`?
+
+#### Source-Only Builds
+
+For source-only builds, you're going to need to build all the source, including the source from the 1xx branch in order to produce a release from a 2xx branch.
+
+There are a variety of scenarios to consider:
+
+1. Has the distro maintainer produced any 1xx assets before from the previous release?
+    * If yes:
+      * Build 1xx branch using artifacts and SDK from previous 1xx release as input
+    * If no:
+      * Build 1xx branch using Microsoft artifacts and SDK
+      * Rebuild 1xx branch using output of first build as input
+1. Is this the initial release of the feature band (e.g. 10.0.200)?
+    * If yes:
+      * Build 2xx branch using output of 1xx artifacts and SDK as input
+    * If no (e.g. 10.0.201):
+      * Has the distro maintainer produced any 2xx assets before from the previous release?
+        * If yes:
+          * Build 2xx branch using artifacts and SDK from previous 2xx release and the shared component 1xx artifacts as input
+        * If no:
+          * Build 2xx branch using Microsoft 2xx artifacts and shared component 1xx artifacts as input
+          * Rebuild 2xx branch using output of first build as input
+
+The assertion with these scenarios is that a minimum of 2 and maximum of 4 builds are required to produce source built output for a 2xx feature band. It will never be necessary to depend on any other feature band besides 1xx and the target feature band. For example, it would never be necessary for a distro maintainer to build the 3xx feature band in order to source build the 4xx feature band.
+
+In order to facilitate the actions that will need to be taken by a distro maintainer or developer, new scripts should be provided (or existing ones updated). Specifically, provide a way to combine the output artifacts from multiple builds to be used as input. An alternative to combining artifacts is to have separate tarball inputs. See the implications of this design choice in [poison leak detection](#poison-leak-detection).
+
+❓ Open Question:
+
+Should a distro maintainer be required to build the *entire* 1xx branch VMR (including the SDK)? That's easier to implement but also potentially wasteful if they're not going to be producing a release for that feature branch. Otherwise, there'd need to be some way to filter which repos get built such that only the repos excluded from the 2xx branch get built. Including all packages, including tools, rather than just shared components, has implications for [poison leak detection](#poison-leak-detection) which need to be considered.
+
```

Comment:
It would be good to add clarity around build command requirements. I think `OfficialBuildId` becomes mandatory if building 2xx or higher branches. Versioning packages with `-ci` or `-dev` would not be enough, to differentiate between those built in 1xx (and brought via 1xx PSB archive) and packages build in 2xx + - this will only affect SDK/Roslyn packages.